### PR TITLE
ViewHotReload moved to the main repo

### PR DESCRIPTION
**Pull request by tomasherceg**
Before merging, there are still two things we need to resolve:

* I've kept the namespace `DotVVM.Diagnostics.ViewHotReload`, but I think we should change it to `DotVVM.Tools`.
* Also, I wonder about the name - should it be just `HotReload`, or `ViewHotReload`? Right now, it doesn't do any hot reload for C# code - it reloads just DotHTML views. However, in the future, we may allow to update the caches when a type is changed (e. g. new properties added to the viewmodel class), so it doesn't concern only the views.

### CODE REVIEW

**Comment by cafour** [comment]
Both the namespace change and the `ViewHotReload` -> `HotReload` rename seem like a good idea to me.

**Comment by tomasherceg** [comment]
We've discussed the name of the package and it should be `DotVVM.HotReload`.

**Comment by quigamdev** [comment]
@tomasherceg  Is this PR ready for review? As I have been told, there are should be some other changes that you want to push.

**Review by exyi — APPROVED** [review]

File: src/Tools/HotReload/Common/HotReloadErrorPageExtension.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public class HotReloadErrorPageExtension : IErrorPageExtension
+    {
+
+        public string GetHeadContents(IDotvvmRequestContext context, Exception ex)
+        {
+            using var textWriter = new StringWriter();
+            var writer = new HtmlWriter(textWriter, context);
+            var renderedResources = new HashSet<string>();
+
+            RenderResource(context, "dotvvm-hotreload", writer, renderedResources);
```
[the remark this answers is not in the document]
Comment (tomasherceg) [reply src/Tools/HotReload/Common/HotReloadErrorPageExtension.cs:21]: There is an exception in RenderResource method that specifically excludes `dotvvm` from the error page. 
The script works without `window.dotvvm` but it has dependencies to SignalR (and jQuery on OWIN - old SignalR needs it).

**Review by quigamdev — APPROVED** [review]

**Pull request merged**